### PR TITLE
Use absolute uris in oidc-client discovery and authorization webclient

### LINKED ISSUE #14541: Check if OIDC Client can use Vertx Web Client more effectively
**Description**
@sd-f has helped with setting the web client options correctly.

What would be good to check next is look at `.getAbs()` (vs `.get()`). @vsevel uses `.getAbs()` in his PR.
The question I have, what is cheaper. If we set the default host and port options then may be it is more efficient to use only the path component of the auth server url and also the token url ? At the moment we set the default host/port - but also use absolute URIs with the `.get()` and `post()`. It feels like we can further optimize something :-).

Also minor optimization is possible by replace `transformToUni` with just `transform` in `OidcClientRecorder` where `JsonObject` response is picked up

### PR DESCRIPTION
**Pull request by sd-f**
Fixes #14541

### CODE REVIEW

**Comment by sd-f** [comment]
@sberyozkin do you mind reviewing once more?

File: extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java [hunk 1]
```diff
@@ -221,13 +192,13 @@ private static void setGrantClientParams(OidcClientConfig oidcConfig, MultiMap g
...
-        return client.get(discoveryUrl).send().onItem().transformToUni(resp -> {
+        return client.getAbs(discoveryUrl).send().onItem().transform(resp -> {
             if (resp.statusCode() == 200) {
                 JsonObject json = resp.bodyAsJsonObject();
-                return Uni.createFrom().item(json.getString("token_endpoint"));
+                return json.getString("token_endpoint");
             } else {
                 LOG.tracef("Discovery has failed, status code: %d", resp.statusCode());
-                return Uni.createFrom().nullItem();
+                throw new OidcClientException("Token endpoint discovery has failed");
```
Comment (sberyozkin) [inline extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java:201]: @sd-f This null is taken care of a bit later [here](https://github.com/quarkusio/quarkus/blob/351778d3ce0aa8a44cc45328ebdc44c293d2f1bb/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java#L119).
We can throw the exception now but then we'd have to refactor the uni code catching it, so if you don't mind then lets keep retuning null here :-)

Comment (sd-f) [reply extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java]: @sberyozkin will be done :) i mixed up the apply with outside call where exceptions are handled.

**Review by sberyozkin — APPROVED** [review]
@sd-f - great - thanks for a super quick PR :-)
Please keep stressing `quarkus-oidc-client` :-)

Thanks

**Comment by sberyozkin** [comment]
@sd-f I've updated the PR description to resolve the linked issue, we can open new ones going forward :-)
thanks

**Pull request merged**